recommence/Checkpoint.py

```diff
@@ -5,33 +5,44 @@
 
 T = TypeVar('T')
 class Checkpoint:
-    def __init__(self, save_path: str):
+    def __init__(self, save_path: str, no_fail: bool = False) -> None:
         self.save_path: str = save_path
         self._data_path: str = f'{save_path}/data.pkl'
         self._data: Dict[str, Any] = {}
 
+        self.no_fail: bool = no_fail  # If true, do not fail if there are issues
         self._load_if_exists()
 
     def __getitem__(self, name: str) -> Any:
         return self._data[name]
 
     def __setitem__(self, name: str, value: Callable[[], T]) -> T:
-        self._data[name] = value()
+        try:
```

Comment:
I suspect there is no way not to fail in this case. I.e. if we can't set `self._data[name]` here, then we can't return it on line 25 below either.

In this case, I actually wouldn't wrap this function in a try/except block and would allow this to fail at runtime. The only way this block would fail is if `value()` fails --- which is on the user (notably, also, this would fail regardless of loading from checkpoint or not, so isn't within our purview).